### ISSUE TITLE
Checks for required configuration don't check thing configuration

### ISSUE THREAD

**Issue by cdjackson** [eclipse/smarthome#5180]
As seen [here](https://github.com/openhab/org.openhab.binding.zwave/issues/832#issuecomment-370136289), if thing configuration is defined in a ConfigProvider that is providing ```thing``` configuration (as opposed to ```thing-type``` configuration) then the checks for the required config parameters does not work properly on thing initialisation.

**Comment by sjka** [comment]
Indeed, the ThingManager is using the config description URI of the ThingType/ChannelType, not their instances.
